### PR TITLE
Add keycloak token verification

### PR DESCRIPTION
**Pull request by Derstilon**

### CODE REVIEW

File: src/services/AuthService.tsx [hunk 1]
```diff
@@ -177,52 +178,66 @@ const Auth = ({ children }: GenericContextProviderProps) => {
...
+			token: KeycloakTokenParsed | undefined
+		) => Promise<void> = token => {
+			const yaptideServices = ['PLG_YAPTIDE_ACCESS'];
 
-		/**
-		 * TODO: Check if user is authorized to use this application (e.g. by checking if user is in a certain group)
-		 */
-		const validUser = true;
+			const validUser = yaptideServices.every(
+				service => token?.plgridAccessServices?.includes(service)
```
[the remark this answers is not in the document]
Comment (Derstilon) [reply src/services/AuthService.tsx:187]: `?` changes the whole expression to undefined if the property before it is undefined. Undefined is a falsy value so everything will be working as intended.

[the remark this answers is not in the document]
Comment (grzanka) [reply src/services/AuthService.tsx:187]: OK, thanks !

[the remark this answers is not in the document]
Comment (grzanka) [reply src/services/AuthService.tsx:187]: I think that `token.plgridAccessServices` is a string, not a list of strings. How `includes` work if we apply on two strings ?

[the remark this answers is not in the document]
Comment (grzanka) [reply src/services/AuthService.tsx:187]: I think that `token.plgridAccessServices` is a string, not a list of strings. How `includes` work if we apply on two strings ?

**Comment by grzanka** [comment]
Seems to work after changing hardcoded service name to some absurd value:

image

**Review by grzanka — APPROVED** [review]

**Pull request merged**